crates/tiny/src/config.rs

```diff
@@ -60,34 +62,231 @@ pub(crate) struct Defaults {
     pub(crate) tls: bool,
 }
 
+// TODO FIXME: I don't understand why we need `Default` bound here on `P`
 #[derive(Deserialize)]
-pub(crate) struct Config {
-    pub(crate) servers: Vec<Server>,
+pub(crate) struct Config<P: Default> {
```

Comment:
For some reason I can't remove `P: Default` here if I want `derive(Deserialize)`, even though `Default` impl for `P` should not be necessary for deserialization of this type (see also line 92 below). @trevarj do you know why? I wonder if this is a serde bug.

Reply:
That doesn't seem right. I don't know enough about what serde expects to comment on it though.

This seems to resolve the problem, right?
```suggestion
pub(crate) struct Config<P> {
    #[serde(default)]
    pub(crate) servers: Vec<Server<P>>,
```

Reply:
Could it be that the heuristics it uses does not or cannot correctly find the trait bound? https://serde.rs/attr-bound.html

Reply:
Fixed this by replacing serde's derived trait bounds : https://github.com/trevarj/tiny/commit/b3a22b265564d7609551e8c6c7249fb0275597e3

Reply:
@osa1 let me know if you want me to PR this change, if it's acceptable. Then maybe we can finish this up

Reply:
@trevarj it would be great if you could take this over!

Reply:
@trevarj 's suggestion worked, marking as resolved.